server/modules/user/service/userService.js

```diff
@@ -62,33 +60,29 @@ export const inviteUser = async (user, surveyId, surveyCycleKey, email, groupUui
     })
   } else {
     await db.tx(async t => {
-      try {
-        const password = passwordGenerator.generate({
-          length: 8,
-          numbers: true,
-          uppercase: true,
-          strict: true,
-        })
-        // Add user to cognito pool
-        const {
-          User: { Username: userUuid },
-        } = await aws.inviteUser(email, password)
-        // Add user to db
-        await UserManager.insertUser(user, surveyId, surveyCycleKey, userUuid, email, groupUuid, t)
-        // Send email
-        const msgParams = {
-          serverUrl,
-          email,
-          password,
-          surveyLabel,
-          groupLabel,
-          temporaryPasswordMsg: '$t(emails.userInvite.temporaryPasswordMsg)',
-        }
-        await Mailer.sendEmail(email, 'emails.userInvite', msgParams, lang)
-      } catch (error) {
-        await aws.deleteUser(email)
-        throw error
+      // Add user to db
+      const password = UserPasswordUtils.generatePassword()
+      const passwordEncrypted = await UserPasswordUtils.encryptPassword(password)
+      await UserManager.insertUser(
+        user,
+        surveyId,
+        surveyCycleKey,
+        email,
+        passwordEncrypted,
+        User.userStatus.INVITED,
+        groupUuid,
+        t,
+      )
+      // Send email
+      const msgParams = {
+        serverUrl,
+        email,
+        password,
+        surveyLabel,
+        groupLabel,
+        temporaryPasswordMsg: '$t(emails.userInvite.temporaryPasswordMsg)',
       }
+      await Mailer.sendEmail(email, 'emails.userInvite', msgParams, lang)
```

Comment:
test if Mailer.sendEmail fails the user is removed from db

Reply:
yes, Mailer.sendEmail is invoked in the transaction block, the transaction rolls back correctly